addons/account/static/tests/x2many_buttons_tests.js

```diff
@@ -0,0 +1,155 @@
+/** @odoo-module **/
+import {click, editInput, getFixture, patchWithCleanup } from "@web/../tests/helpers/utils";
+import { makeView, setupViewRegistries } from "@web/../tests/views/helpers";
+
+
+let serverData;
+let target;
+let fromArch = `
+    <form>
+        <field name="display_name"/>
+        <field name="ref"/>
+        <field name="duplicated_ref_ids" widget="x2many_buttons"/>
+    </form>
+`;
+
+QUnit.module("Fields", (hooks) => {
+    hooks.beforeEach(() => {
+        target = getFixture();
+        serverData = {
+            models: {
+                "account.move": {
+                    fields: {
+                        display_name: {
+                            "string": "Display Name",
+                            "type": "string"
+                        },
+                        duplicated_ref_ids: {
+                            "string": "Duplicated Bills",
+                            "type": "many2many",
+                            "relation": "account.move"
+                        },
+                        ref: {
+                            "string": "Bill Reference",
+                            "type": "char"
+                        }
+                    },
+                    records: [
+                        // for the sake of mocking data, we don't care about the consistency of duplicated refs across records
+                        { id: 1, display_name: "Bill 1", duplicated_ref_ids: [2, 3] , ref: "b1"},
+                        { id: 2, display_name: "Bill 2", duplicated_ref_ids: [1] , ref: "b2"},
+                        { id: 3, display_name: "Bill 3", duplicated_ref_ids: [1] , ref: "b3"},
+                        { id: 4, display_name: "Bill 4", duplicated_ref_ids: [1,2,3] , ref: "b4"},
+                        { id: 5, display_name: "Bill 5", duplicated_ref_ids: [] , ref: "b5"},
+                        { id: 6, display_name: "Bill 6", duplicated_ref_ids: [1,2,3,4,5] , ref: "b6"},
+                    ],
+                }
+            }
+        };
+        setupViewRegistries();
+    });
```

Comment:
- All 3 variables could be declared inside the module scope
- serverData and the arch could be a const
```suggestion
QUnit.module("Fields", (hooks) => {
    const serverData = {
            models: {
                "account.move": {
                    fields: {
                        display_name: {
                            "string": "Display Name",
                            "type": "string"
                        },
                        duplicated_ref_ids: {
                            "string": "Duplicated Bills",
                            "type": "many2many",
                            "relation": "account.move"
                        },
                        ref: {
                            "string": "Bill Reference",
                            "type": "char"
                        }
                    },
                    records: [
                        // for the sake of mocking data, we don't care about the consistency of duplicated refs across records
                        { id: 1, display_name: "Bill 1", duplicated_ref_ids: [2, 3] , ref: "b1"},
                        { id: 2, display_name: "Bill 2", duplicated_ref_ids: [1] , ref: "b2"},
                        { id: 3, display_name: "Bill 3", duplicated_ref_ids: [1] , ref: "b3"},
                        { id: 4, display_name: "Bill 4", duplicated_ref_ids: [1, 2, 3] , ref: "b4"},
                        { id: 5, display_name: "Bill 5", duplicated_ref_ids: [] , ref: "b5"},
                        { id: 6, display_name: "Bill 6", duplicated_ref_ids: [1, 2, 3, 4, 5] , ref: "b6"},
                    ],
                }
            }
        };
    const fromArch = `
        <form>
            <field name="display_name"/>
            <field name="ref"/>
            <field name="duplicated_ref_ids" widget="x2many_buttons"/>
        </form>
    `;
    let target;
    hooks.beforeEach(() => {
        target = getFixture();
        setupViewRegistries();
    });
```